x-pack/plugins/uptime/public/components/monitor/synthetics/step_detail/use_step_waterfall_metrics.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useSelector } from 'react-redux';
+import { createEsParams, useEsSearch } from '../../../../../../observability/public';
+import { selectDynamicSettings } from '../../../../state/selectors';
+import { MarkerItems } from '../waterfall/context/waterfall_chart';
+
+export interface Props {
+  checkGroup: string;
+}
+
+export const useStepWaterfallMetrics = ({ checkGroup }: Props) => {
+  const { settings } = useSelector(selectDynamicSettings);
+
+  const heartbeatIndices = settings?.heartbeatIndices || '';
+
+  const { data, loading } = useEsSearch(
+    createEsParams({
+      index: heartbeatIndices!,
+      body: {
+        query: {
+          bool: {
+            filter: [
+              {
+                term: {
+                  'monitor.check_group': checkGroup,
+                },
+              },
+              {
+                term: {
+                  'synthetics.type': 'journey/metrics',
+                },
+              },
+            ],
+          },
+        },
+        fields: ['browser.*'],
+        _source: false,
+      },
+    }),
+    [heartbeatIndices, checkGroup]
+  );
+
+  const metrics: MarkerItems = [];
+
+  if (data) {
+    const metricDocs = (data.hits.hits as unknown) as Array<{ fields: any }>;
+    let navigationStart = 0;
```

Comment:
FYI, This is prone to errors and wold result in absurd values when `navigationStart` is not present in the doc. 

I would change this logic to this.
- Find the navigationStart from the metricDocs and if not present don't calculate the flyout metrics. 
- If present, calculate the offset metrics as already done.

Reply:
Improved it to make sure check for existence explicitly.